### ISSUE TITLE
interactive line profile plots in a single figure for comparison 

### ISSUE THREAD

**Issue by muratyesibolati** [hyperspy/hyperspy#2335]
Hi all, I have a question regarding Line2DRoi and interactive plot. there are two images, and I would like to have a lineplot in each image with the same coordinates, then make line profile plots ( two profile from each image) to a SINGLE figure. this can be done with Line2DROoi, and the lineplots on image are asynchronized, and the interactive makes it possible to drag the lineplot. at the end you have two line profiles in two separate figure, and difficult to compare. however, I would like to have this interactive lineprofile plots in a SINGLE figure for comparison, at the same time being interactive. I tried with plot_spectra, you can add plots, however, it is not interactive. could someone suggest any? thank you in advance.

here is a code 
line_roi = hs.roi.Line2DROI(0, 0, 512, 512, 1)
Holo1.plot()
Holo1R.plot()
roi1D = line_roi.interactive(Holo1, color="yellow")
roi1D_R = line_roi.interactive(Holo1R, color="green")
roi1D.plot()
roi1D_R.plot()
hs.plot.plot_spectra([roi1D,roi1D_R])

**Comment by ericpre** [comment]
Indeed, it would be nice to be able to do this with the `plot_spectra` function directly.

Using the hyperspy events, it works actually quite nicely:

```python
import hyperspy.api as hs
from functools import partial

ima0 = hs.datasets.example_signals.object_hologram()
ima1 = hs.datasets.example_signals.reference_hologram()

ima0.plot()
ima1.plot()
color = ["tab:blue", "tab:orange"]

line_roi = hs.roi.Line2DROI(0, 0, 256, 256, 1)
roi0 = line_roi.interactive(ima0, color=color[0])
roi1 = line_roi.interactive(ima1, color=color[1])

ax = hs.plot.plot_spectra([roi0, roi1], color=color)

# Get the two matplotlib lines to update:
line0 = ax.get_lines()[0]
line1 = ax.get_lines()[1]

def update_line(roi, line):
    x_axis = roi.axes_manager[-1].axis
    line.set_data(x_axis, roi.data)
    fig = line.get_figure()
    ax = fig.get_axes()[0]
    ax.set_xlim(x_axis[0], x_axis[-1])
    fig.canvas.draw()

# Get the functions to update the two lines:
connected_line0 = partial(update_line, roi=roi0, line=line0)
connected_line1 = partial(update_line, roi=roi1, line=line1)

# connect the functions to the ROIs `data_changed` event
roi0.events.data_changed.connect(connected_line0, [])
roi1.events.data_changed.connect(connected_line1, [])

```

**Comment by muratyesibolati** [comment]
HI Eric, 
this is awesome solution, I could not implement by myself.
it worked pretty well. thanks for the help.

one very tiny addup, y scale does not update automatically, and pyplot autoscale did not work. I just add one more sentence to the code. the code looks long,  but it works.  :) 

    ax.set_ylim(min(min(roi0.data),min(roi1.data)),max(max(roi0.data),max(roi1.data)))

**Comment by ericpre** [comment]
For completeness, after #2414 (available since hyperspy 1.6), the example given above can now be simplify to: 

```python
import hyperspy.api as hs
from functools import partial

ima0 = hs.datasets.example_signals.object_hologram()
ima1 = hs.datasets.example_signals.reference_hologram()

ima0.plot()
ima1.plot()
color = ["tab:blue", "tab:orange"]

line_roi = hs.roi.Line2DROI(0, 0, 256, 256, 1)
roi0 = line_roi.interactive(ima0, color=color[0])
roi1 = line_roi.interactive(ima1, color=color[1])

ax = hs.plot.plot_spectra([roi0, roi1], color=color)
```
